### DISCUSSION TITLE
open-files.cc:160 Couldn't create '/mnt/sdb/transmission/downloads': File exists (17)

### DISCUSSION THREAD

**Discussion by stephenboston** [transmission/transmission#8430] [Support]
### What is the issue?

I have a VirtualBox setup for torrents like this:

Both host and guest are Arch linux on `6.8.6-arch1-1` kernel.

On the host, we have an external drive`/dev/sdb` mounted to `/mnt/sdb`, The mount is owned by `transmission:transmission`.

On the guest we have a sshfs mount of `host:/mnt/sdb` to `guest:/mnt/sdb`  (Shared folders not working)
Permissions are identical on both sides and **uid/gid for transmission are the same value**.  

On the guest, `/var/lib/transmission/.config/tranmission-remote/settings.json` sets the download directory to `/mnt/sdb/transmission/downloads `

I can add a magnet link through the web-ui. The torrent is started, metadata retrieved, but then  we get the error :

     ERR open-files.cc:160 Couldn't create '/mnt/sdb/transmission/downloads': File exists (17) (open-files.cc:160)

transmission is started by root through a systemd service in /lib/systemd/system/transmssion.service 

        
        [Unit]
        Description=Transmission BitTorrent Daemon
        Wants=network-online.target
        After=network-online.target
        
        [Service]
        User=transmission
        Type=notify
        ExecStart=/usr/bin/transmission-daemon -f --log-level=error
        ExecReload=/bin/kill -s HUP $MAINPID
        NoNewPrivileges=true
        MemoryDenyWriteExecute=true
        ProtectSystem=true
        PrivateTmp=true
        
        [Install]
        WantedBy=multi-user.target
        

### Which application of Transmission?

transmission-remote

### Which version of Transmission?

transmission-cli 4.0.5-1

**Comment by killemov** [comment]
The systemd service file doesn't look very interesting. Am I correct to assume your external drive does not have a partition table but a single flat partition? Can you manually write anything in that folder from your VirtualBox guest environment as user transmission? If no, then at least it is not a transmission problem.

**Comment by stephenboston** [comment]
Your assumption is correct. 

I can create and delete files and directories in the /mnt/sdb/transmission directory where 'I'  is stephen, a member of transmission group.

Something I forgot to say: the `.config..torrents` and `.config..resume` directories are symlinks to `/mnt/sdb/transmission/torrents` and `/mnt/sdb/transmission/resume`. I do this because I want to version the config but not the data.  Problem?

**Comment by killemov** [comment]
> I can create and delete files and directories in the /mnt/sdb/transmission directory where 'I' is stephen, a member of transmission group.

You really need to test this from your VirtualBox environment as user transmission. Create /mnt/sdb/transmission/downloads as transmission, chmod 777 it, and let transmission-daemon(?) run again.

`.config..torrents` and `.config..resume`  don't mean anything to me.

Under Debian the two folders `resume` and `torrents` are located in `/var/lib/transmission-daemon/.config/transmission-daemon` which also seems weird but works.
